renderer-macos/lib/components/ScrollView.re

```diff
@@ -0,0 +1,51 @@
+open Brisk;
+open Layout;
+
+type attr = [ View.attr];
+
+type style = list(attr);
+
+module ScrollableArea = {
+  let component = nativeComponent("ScrollView");
+
+  let make = (~style: style=[], children: Brisk.syntheticElement) =>
+    component((_: Hooks.empty) =>
+      {
+        make: () => {
+          let view = BriskScrollView.make();
+          {view, layoutNode: makeLayoutNode(~style, view)};
+        },
+        configureInstance: (~isFirstRender as _, {view} as node) => {
+          style
+          |> List.iter(attr =>
+               switch (attr) {
+               | #Styles.viewStyle => Styles.setViewStyle(view, attr)
+               | #Layout.style => ()
+               }
+             );
+          node;
+        },
+        children,
+      }
+    );
+
+  let createElement = (~style=[], ~children, ()) =>
+    element(make(~style, listToElement(children)));
+};
+
+let component = component("ScrollView");
+
+let make = (~style: style=[], children: Brisk.syntheticElement) =>
+  component((_: Hooks.empty) => {
+    let scrollStyle = [flexGrow(1.), overflow(`Scroll)];
+    let contentStyle = [position(~top=0., ~left=0., ~right=0., `Absolute)];
+
+    <View style>
+      <ScrollableArea style=scrollStyle>
+        <View style=contentStyle> children </View>
```

Comment:
The fundamental problem is that `flex` doesn't always apply seamlessly to `Cocoa`.

We can only set a single view to `NSScrollView`, so we have to provide a container View for our children _whether it's react elements or cocoa instances hierarchy_. Having it on Objc side lead me nowhere as it showed promise of even more hacks than this, IMO, simple and elegant solution.

Reply:
I don't think it doesn't apply seamlessly.

This is the opposite to a solution - if you pass two children to `ScrollableArea` the abstraction breaks. This component sets precedence for other cases like this in the future so it's especially important to get this right.

Reply:
Idea, add child node, inside `makeLayoutNode` whose context will be `documentView` with the desired styles.

Reply:
I might but sadly not today. It should not be too hard though.
1. create two nodes - one for the outer view (i.e. scroll view), second for the contentView/documentView
2. add the second one as a child to the first one
3. When you `insertNode` to the scroll view append to the documentView and as child node's child
3. When you insert `scrollView` as a child, remove the scrollView itself. I think it might be a pretty common case in the future.

Reply:
I don't see it as a blocker, we can track an issue for this.

Reply:
I definitely see this as a blocker, allowing leaky abstractions is the way to hacks on top of hacks. We cannot care about the structure of the repo more than we do about the way it's built.